### ISSUE TITLE
[ENH] Random walk forecaster

### ISSUE THREAD

**Issue by aiwalter** [sktime/sktime#3834]
**Is your feature request related to a problem? Please describe.**
Would be nice to have a random walk forecaster for the sake of benchmarking other forecasters with.

**Describe the solution you'd like**
we can either adapt it from `statsforecast` [implementation](https://nixtla.github.io/statsforecast/models.html#randomwalkwithdrift) or implement is ourselves.

**Comment by RNKuhns** [comment]
Optimal forecast for Random Walk is naive (last value). Or naive with drift in case of trend. We have the first implemented and the second can be completed using pipeline.

**Comment by aiwalter** [comment]
Yes, but people might want to use actually random values with different seeds?

**Comment by RNKuhns** [comment]
Wouldn’t using random values would be part of the simulation API?

**Comment by fkiraly** [comment]
agree with @RNKuhns on point forecasts, and on the point that random walk would be more part of a simulation interface along the lines that @lmmentel is proposing.

For probabilistic forecasts, instead of simulating you would probably instead forecast appropriately widening distributions.

**Comment by ltsaprounis** [comment]
Yes - you can achieve a "random" forecast by doing a single simulation under the proposed simulation API in #3462.

**Comment by RNKuhns** [comment]
> Yes - you can achieve a "random" forecast by doing a single simulation under the proposed simulation API in #3462.

Makes sense as part of the simulation API, and also functionality to generate example datasets but I don’t think a separate forecaster makes sense.
